moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -129,6 +129,15 @@ bool RuckigSmoothing::applySmoothing(robot_trajectory::RobotTrajectory& trajecto
 
       getNextRuckigInput(ruckig_output, next_waypoint, num_dof, idx, ruckig_input);
 
+      // Skip repeated waypoints with no change in position. Ruckig does not handle this well and there's really no
+      // need to smooth it. Simply leave it equal to the previous (identical) waypoint.
+      bool identical_waypoint =
+          checkForIdenticalWaypoints(*trajectory.getWayPointPtr(waypoint_idx), *next_waypoint, trajectory.getGroup());
```

Comment:
This really only compares the position. Further down in the loop Ruckig overwrites the position and dynamics in the same trajectory just to produce a new output that may have different positions and dynamics for the next waypoint. Wouldn't skipped waypoints lead to steps in the trajectory that are not smoothed at all? I think it would be safer to either make sure the robot is really halting (dynamics == 0) or remove duplicate waypoints if the time stamp is also the same.

Reply:
Nit: I would probably skip the bool variable

Reply:
I changed this to remove duplicated waypoints just once at the very beginning. That's more efficient. Good idea.

I think the waypoints should be removed even if (dynamics != 0) because they cause circular motions otherwise.